src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -390,35 +400,38 @@ private MsalAccessTokenCacheItem FindAccessTokenCommon
             IEnumerable<MsalAccessTokenCacheItem> filteredItems =
                 tokenCacheItems.Where(item => ScopeHelper.ScopeContains(item.ScopeSet, requestParams.Scope));
 
-            requestParams.RequestContext.Logger.Info("Matching entry count after filtering by scopes - " + filteredItems.Count());
+            IEnumerable<MsalAccessTokenCacheItem> msalAccessTokenCacheItems = filteredItems.ToList();
```

Comment:
>ToList() [](start = 92, length = 8)

you are adding a few ToList operations. Are those strictly needed? It will have some performance impact if needed.

Reply:
Agreed, the entire gigantic method uses .ToList() everywhere. Would be good to remove all of that and just IEnumerable and Count().

Reply:
this was a resharper suggestion...i reverted back.

---
In reply to: [238931452](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/708#discussion_r238931452) [](ancestors = 238931452)

Reply:
you mean use Count() instead of ToList() in all places? 

---
In reply to: [239032426](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/708#discussion_r239032426) [](ancestors = 239032426)